src/styles/index.scss

```diff
@@ -1,3 +1,83 @@
+$day-size: 100px;
+$gap-size: 1px;
+$font-size: 30px;
+$day-bg: #eee;
+$hover-bg: #ffbfcb;
+$days: (
+  mon: 1,
+  tue: 2,
+  wed: 3,
+  thu: 4,
+  fri: 5,
+  sat: 6,
+  sun: 7,
+);
+
 body {
+  height: 100vh;
   margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  display: grid;
+  grid-template-columns: repeat(7, $day-size);
+  grid-auto-rows: $day-size;
+  gap: $gap-size;
+
+  @for $length from 28 through 30 {
+    &--month-length-#{$length} .calendar__day:nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
```

Comment:
The loop to handle the month lengths can be simplified. Since you already have a loop that iterates through 31 days for the `.calendar__day` class, you can use the `:nth-last-child()` selector to hide the excess days, which would make this loop redundant.